velox/exec/StreamingAggregation.cpp

```diff
@@ -341,15 +345,20 @@ RowVectorPtr StreamingAggregation::getOutput() {
   masks_->addInput(input_, inputRows_);
 
   const auto numPrevGroups = numGroups_;
-
   assignGroups();
   initializeNewGroups(numPrevGroups);
   evaluateAggregates();
 
   RowVectorPtr output;
-  if (numGroups_ > minOutputBatchSize_) {
-    output = createOutput(
-        std::min(numGroups_ - 1, static_cast<size_t>(maxOutputBatchSize_)));
+  if (trySplitOutputAtInputBoundary_) {
+    if ((numPrevGroups != 0) && (numGroups_ > minOutputBatchSize_)) {
+      output = createOutput(std::min(numGroups_ - 1, numPrevGroups));
```

Comment:
Could there be any case that the last group of prev groups is partially filled. This getOutput call uses the current input_ to fill out that group, and a few new groups. Then the current result batch is not strictly cut by input boundary?